src/JsonSchema/Uri/UriRetriever.php

```diff
@@ -140,10 +125,10 @@ public function retrieve($uri, $baseUri = null)
         $resolvedUri = $fetchUri = $resolver->resolve($uri, $baseUri);
```

Comment:
I think that support for `$baseUri` should be dropped from this method. Always passing URIs makes the code much clearer and removes the need to construct a `UriResolver` upon every call.

The `UriResolver` can be created (and cached) in the calling code.